consensus/src/execution_ctx.rs

```diff
@@ -31,10 +31,40 @@ use tokio::time;
 use tokio::time::Instant;
 use tracing::{debug, error, info, trace};
 
+/// A pool of all generated committees
+#[derive(Default)]
+pub struct RoundCommittees {
+    committees: HashMap<u8, Committee>,
+}
```

Comment:
This is way too similar to `CommitteeSet`. Probably in the future we can simplify that and use a single cache